src/Surfnet/StepupGateway/GatewayBundle/Saml/ResponseContext.php

```diff
@@ -162,7 +167,9 @@ public function saveAssertion(SAML2_Assertion $assertion)
 
         // same for the entityId of the authenticating Authority
         $authenticatingAuthorities = $assertion->getAuthenticatingAuthority();
-        if (!empty($authenticatingAuthorities)) {
+        if (empty($authenticatingAuthorities)) {
+            $this->stateHandler->setAuthenticatingIdp($assertion->getIssuer());
+        } else {
             $this->stateHandler->setAuthenticatingIdp(reset($authenticatingAuthorities));
```

Comment:
Afaik, we should not do this. If SSP is an issue (dev?) we should fix that :wink: